crates/net/network/src/eth_requests.rs

```diff
@@ -185,6 +185,32 @@ where
 
         let _ = response.send(Ok(BlockBodies(bodies)));
     }
+
+    fn on_receipts_request(
+        &mut self,
+        _peer_id: PeerId,
+        request: GetReceipts,
+        response: oneshot::Sender<RequestResult<Receipts>>,
+    ) {
+        let mut receipts = Vec::new();
+
+        for hash in request.0 {
+            if let Some(receipts_by_block) =
+                self.client.receipts_by_block(BlockHashOrNumber::Hash(hash)).unwrap_or_default()
+            {
+                receipts.push(
+                    receipts_by_block
+                        .into_iter()
+                        .map(|receipt| receipt.with_bloom())
+                        .collect::<Vec<_>>(),
+                );
+            } else {
+                break
+            }
```

Comment:
we also need to enforce the soft limit here which is 2MB

because we currently don't have a way of sending raw receipts data we need to do some approximations, see bodies request handler for example

this is actually something we def need to fix